build_runner/test/daemon/daemon_test.dart

```diff
@@ -143,6 +143,19 @@ main() {
       ]).create();
     });
 
+    test('supports --enable-experiment option', () async {
+      await _startDaemon(options: ['--enable-experiment=fake-experiment']);
```

Comment:
Before it would fail due to unrecognized option. I am not really sure why compilation does not fail, do I need to pass the options somewhere else?

Reply:
> Before it would fail due to unrecognized option.

I am pretty sure option did exist before - it just wasn't being used. So I am pretty sure this would pass before and after.

> I am not really sure why compilation does not fail, do I need to pass the options somewhere else?

It could be that this build doesn't actually do compilation? Or maybe we only emit a warning in this case and the build doesn't fail. Can you check the logs to see if there is a warning?

Reply:
Yeah your are right - I think we are not passing option down to the build, but I am not sure where that would be. The build does run compilation, I verified by adding code that uses constructor tearoffs and enabling the experiment. Compilation failed... I'd appreciate any hints :)

Reply:
Update - I forgot to update the language constrain in the test file. Apparently my change does work!
Still, no warning in the logs if I misspell the option. I see the warning if i run `dart --enable-experiment=blah` but nor when i run `dart run build_runner build --enable-experiment=blah`. Looks like we are losing the warnings in the build runner?

Reply:
there is an `_test/test/build_integration_test/dart` that manages to find `'Unknown experiment: fake-experiment'` in the stdout of build_runner but it does it seems to do so because of saving summaries, which I guess my test does not do:
```
[SEVERE] build_vm_compilers:vm on package:test_process/test_process.vm.module (cached):
Error creating test_process|lib/test_process.vm.dill
Error creating kernel summary for module:test_process|lib/test_process.vm.dill

Response:--dart-sdk-summary=file:///Users/annagrin/.dart-sdks/2.15.0-68.0.dev/lib/_internal/vm_platform_strong.dill --output=/private/var/folders/gf/lqp6vjc55k1c8q0bwhjd8dg800p2zc/T/scratch_spacetuIqPr/packages/test_process/test_process.vm.dill --packages-file=org-dartlang-app:///.dart_tool/package_config.json --multi-root-scheme=org-dartlang-app --exclude-non-sources --no-summary-only --target=vm --libraries-file=file:///Users/annagrin/.dart-sdks/2.15.0-68.0.dev/lib/libraries.json --enable-experiment=fake-experiment --no-sound-null-safety --source=package:test_process/test_process.dart
Unknown experiment: fake-experiment
#0      initializeCompiler.<anonymous closure> (package:front_end/src/api_unstable/bazel_worker.dart:129:25)
#1      parseExperimentalFlags (package:front_end/src/api_prototype/compiler_options.dart:404:16)
#2      initializeCompiler (package:front_end/src/api_unstable/bazel_worker.dart:127:35)
#3      computeKernel (package:frontend_server/compute_kernel.dart:243:22)
#4      KernelWorker.performRequest (file:///opt/s/w/ir/cache/builder/sdk/utils/bazel/kernel_worker.dart:70:26)
```

Reply:
Ya I have been playing with this a bit as well and I really don't understand why we aren't seeing these same warnings in your build :(.

Reply:
I created a bug for not reporting ddc warnings in build_runner and build_daemon. Somehow we only see them when summaries are created... 

https://github.com/dart-lang/build/issues/3184

Reply:
But the behavior of the tool is different in those cases - in my case ddc just prints something, and when the summary is created above there is an exception thrown, so that's probably why it gets reported.

Reply:
Lol the throwing behavior seems to be a bug in the compiler:)
```
➜  dartdevc --enable-experiment=constructor-tearoff web/main.dart -o output.js -s output.dill
Unknown experiment: constructor-tearoff
We're sorry, you've found a bug in our compiler.
You can report this bug at:
    https://github.com/dart-lang/sdk/issues/labels/web-dev-compiler
Please include the information below in your report, along with
any other information that may help us track it down. Thanks!
-------------------- %< --------------------
    dartdevc -k arguments: --enable-experiment=constructor-tearoff web/main.dart -o output.js -s output.dill
    dart --version: 2.15.0-68.0.dev (dev) (Wed Sep 1 05:51:32 2021 -0700) on "macos_x64"

Crash when compiling org-dartlang-app:/web/main.dart,
at character offset null:
Null check operator used on a null value
#0      generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:101:63)
<asynchronous suspension>
#1      withCrashReporting (package:front_end/src/fasta/crash.dart:121:12)
<asynchronous suspension>
#2      generateKernel.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:51:12)
<asynchronous suspension>
#3      generateKernel (package:front_end/src/kernel_generator_impl.dart:50:10)
<asynchronous suspension>
#4      compile (package:front_end/src/api_unstable/ddc.dart:208:7)
<asynchronous suspension>
#5      _compile (package:dev_compiler/src/kernel/command.dart:338:14)
<asynchronous suspension>
#6      compile (package:dev_compiler/src/kernel/command.dart:51:12)
<asynchronous suspension>
#7      main (file:///opt/s/w/ir/cache/builder/sdk/pkg/dev_compiler/bin/dartdevc.dart:38:18)
<asynchronous suspension>


#0      generateKernelInternal.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:101:63)
<asynchronous suspension>
#1      withCrashReporting (package:front_end/src/fasta/crash.dart:121:12)
<asynchronous suspension>
#2      generateKernel.<anonymous closure> (package:front_end/src/kernel_generator_impl.dart:51:12)
<asynchronous suspension>
#3      generateKernel (package:front_end/src/kernel_generator_impl.dart:50:10)
<asynchronous suspension>
#4      compile (package:front_end/src/api_unstable/ddc.dart:208:7)
<asynchronous suspension>
#5      _compile (package:dev_compiler/src/kernel/command.dart:338:14)
<asynchronous suspension>
#6      compile (package:dev_compiler/src/kernel/command.dart:51:12)
<asynchronous suspension>
#7      main (file:///opt/s/w/ir/cache/builder/sdk/pkg/dev_compiler/bin/dartdevc.dart:38:18)
<asynchronous suspension>
```

Reply:
The compiler bug is unrelated though - we produce summaries using bazel_worker it seems and it throws, as opposed to just running dartdevc that I think is not supposed to throw but just report the warning.

Reply:
Thanks for updating the test! I am resolving this conversation in favor of the new issue.